Packs/ApiModules/Scripts/CrowdStrikeApiModule/CrowdStrikeApiModule.py

```diff
@@ -0,0 +1,186 @@
+from CommonServerPython import *
+from CommonServerUserPython import *
+import requests
+
+
+class CrowdStrikeClient(BaseClient):
+
+    def __init__(self, params):
+        """
+        CrowdStrike Client class that implements OAuth2 authentication.
+        Args:
+            params: Demisto params
+        """
+        self._verify = not params.get('insecure', False)
+        self._username = params.get('credentials', {}).get('identifier')
+        self._password = params.get('credentials', {}).get('password')
+        self._proxy = params.get('proxy', False)
+        self._base_url = "https://api.crowdstrike.com/"
+        self._ok_codes = tuple()  # type: ignore[var-annotated]
+        self._session = requests.Session()
+        self._token = self._generate_token()
+        self._headers = {'Authorization': 'bearer ' + self._token}
+        if not self._proxy:
+            self._session.trust_env = False
+
+        # super().__init__(base_url=self._base_url, verify=self._verify, *args, **kwargs)  # type: ignore[misc]
```

Comment:
You wrote in your client declaration that you are inheriting from the BaseClient but you never did call it. Why are not using it and instead we are coping stuff out of it?